app/src/main/java/org/mozilla/fenix/compose/cfr/CFRPopup.kt

```diff
@@ -0,0 +1,123 @@
+    val indicatorDirection: CFRPopup.IndicatorDirection = CFRPopup.IndicatorDirection.UP,
+    val dismissOnBackPress: Boolean = true,
+    val dismissOnClickOutside: Boolean = true,
+    val overlapAnchor: Boolean = false,
+    val indicatorArrowStartOffset: Dp = CFRPopup.DEFAULT_INDICATOR_START_OFFSET.dp,
+)
+
+/**
+ * CFR - Contextual Feature Recommendation popup.
+ *
+ * @param text [String] shown as the popup content.
+ * @param anchor [View] that will serve as the anchor of the popup and serve as lifecycle owner
+ * for this popup also.
+ * @param properties [CFRPopupProperties] allowing to customize the popup behavior.
+ * @param onDismiss Callback for when the popup is dismissed indicating also if the dismissal
+ * was explicit - by tapping the "X" button or not.
+ * @param action Optional other composable to show just below the popup text.
+ */
+class CFRPopup(
+    private val text: String,
+    private val anchor: View,
+    private val properties: CFRPopupProperties = CFRPopupProperties(),
+    private val onDismiss: (Boolean) -> Unit = {},
+    private val action: @Composable (() -> Unit) = {}
+) {
+    // This is just a facade for the CFRPopupFullScreenLayout composable offering a cleaner API.
+
+    @VisibleForTesting
+    internal var popup: WeakReference<CFRPopupFullScreenLayout>? = null
+
+    /**
+     * Construct and display a styled CFR popup shown at the coordinates of [anchor].
+     * This popup will be dismissed when the user clicks on the "x" button or based on other user actions
+     * with such behavior set in [CFRPopupProperties].
+     */
+    fun show() {
+        anchor.post {
+            CFRPopupFullScreenLayout(text, anchor, properties, onDismiss, action).apply {
+                this.show()
+                popup = WeakReference(this)
+            }
+        }
+    }
+
+    /**
+     * Immediately dismiss this CFR popup.
+     * The [onDismiss] callback won't be fired.
+     */
+    fun dismiss() {
+        popup?.get()?.dismiss()
+    }
+
+    /**
+     * Possible direction for the arrow indicator of a CFR popup.
+     * The direction is expressed in relation with the popup body containing the text.
+     */
+    enum class IndicatorDirection {
+        UP,
+        DOWN
+    }
+
+    companion object {
+        /**
+         * Default width for all CFRs.
+         */
+        internal const val DEFAULT_WIDTH = 335
+
+        /**
+         * Fixed horizontal padding.
+         * Allows the close button to extend with 10dp more to the end and intercept touches to
+         * a bit outside of the popup to ensure it respects a11y recommendations of 48dp size while
+         * also offer a bit more space to the text.
+         */
+        internal const val DEFAULT_HORIZONTAL_PADDING = 10
+
+        /**
+         * How tall the indicator arrow should be.
+         * This will also affect how wide the base of the indicator arrow will be.
+         */
+        internal const val DEFAULT_INDICATOR_HEIGHT = 15
```

Comment:
Should this be 16dp? https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/res/values/dimens.xml#45

Reply:
The arrow drawable that `cfr_triangle_height == 16dp` applies to - [`ic_cfr_triangle`](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/res/drawable/ic_cfr_triangle.xml) actually has a lot of padding so the visual height will be maybe half of that value, while `cfr_triangle_width == 16dp` applied to the same drawable would probably result visually in a width of ~15dp.

![image](https://user-images.githubusercontent.com/11428869/182680976-9d63f25b-c416-45ab-8df9-a5eeefd4c39b.png)

&nbsp;

To me the arrow seemed a bit small in height so I tried a higher default.
Kate just confirmed the designs shown in https://mozilla-hub.atlassian.net/browse/FNXV2-21106 look good so I would say we can go with the value of `15dp` from here and change afterwards if needed.